test-dev-buffer/main.cpp

```diff
@@ -0,0 +1,236 @@
+        exit(1);
+    }
+
+    
+
+    // Create kernel
+    cl_kernel kernel = clCreateKernel(program, "increment", &err);
+    checkError(err, "creating kernel");
+
+    // Create a device buffer with CL_MEM_DEVICE_ADDRESS_EXT flag
+    const size_t buffer_size = 1024 * sizeof(int);
+    cl_mem device_buffer = clCreateBuffer(context, 
+                                        CL_MEM_READ_WRITE | CL_MEM_DEVICE_ADDRESS_EXT,
+                                        buffer_size, nullptr, &err);
+    checkError(err, "creating device buffer");
+
+    // Get device pointer
+    cl_mem_device_address_EXT device_ptr;
+    err = clGetMemObjectInfo(device_buffer, CL_MEM_DEVICE_PTR_EXT, 
+                            sizeof(device_ptr), &device_ptr, nullptr);
+    if (err != CL_SUCCESS) {
+        std::cout << "Failed to get device pointer (Error: " << err << ")\n";
+        return false;
+    }
+    std::cout << "Successfully obtained device pointer: 0x" << std::hex << device_ptr << std::dec << std::endl;
+
+    // Initialize buffer with test data
+    std::vector<int> host_data(1024, 1);  // Initialize with 1s
+    err = clEnqueueWriteBuffer(queue, device_buffer, CL_TRUE, 0, buffer_size, 
+                              host_data.data(), 0, nullptr, nullptr);
+    checkError(err, "writing to buffer");
+
+    // Set kernel argument using the device pointer
+    err = clSetKernelArgDevicePointerEXT(kernel, 0, device_ptr);
+    if (err != CL_SUCCESS) {
+        std::cout << "Failed to set kernel argument with device pointer (Error: " << err << ")\n";
+        return false;
+    }
+
+    // Execute kernel
+    size_t global_size = 1024;
+    err = clEnqueueNDRangeKernel(queue, kernel, 1, nullptr, &global_size, 
+                                nullptr, 0, nullptr, nullptr);
+    checkError(err, "enqueueing kernel");
+
+    // Read back results
+    std::vector<int> result(1024);
+    err = clEnqueueReadBuffer(queue, device_buffer, CL_TRUE, 0, buffer_size, 
+                             result.data(), 0, nullptr, nullptr);
+    checkError(err, "reading results");
+
+    // Verify results
+    bool correct = true;
+    for (int i = 0; i < 1024; i++) {
+        if (result[i] != 2) {  // Should be 1 + 1
+            correct = false;
+            break;
+        }
+    }
+    std::cout << "Computation " << (correct ? "successful" : "failed") << std::endl;
+
+    // Clean up
+    clReleaseMemObject(device_buffer);
+    clReleaseKernel(kernel);
+    clReleaseProgram(program);
+    clReleaseCommandQueue(queue);
+    clReleaseContext(context);
+
+    return correct;
+}
+
+int main() {
+    cl_platform_id platform;
+    cl_device_id device = selectDevice(platform);
+
+    std::cout << "\nRunning kernel test with device address extension...\n";
+    if (!kernelTest(platform, device)) {
+        std::cout << "Kernel test failed!\n";
+        return 1;
+    }
+    
+    std::cout << "All tests completed successfully\n";
+    return 0;
+}
```

Comment:
I'm guessing the intention is to move all the test code to `tests/api/buffer_device_address.cpp`, isn't it?